### PR TITLE
[exporter/splunkhec] Implement maxEventSize on exporter side

### PR DESCRIPTION
**Pull request by VihasMakwana**
**Description:** 
Added a new config to apply the maxEventSize on our data.

- ***It applies this limit on uncompressed data***

**Link to tracking Issue:** [18066](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18066)

**Testing:** Relevant test cases added

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/splunkhecexporter/client.go [hunk 1]
```diff
@@ -221,6 +221,7 @@ func (c *client) pushLogRecords(ctx context.Context, lds plog.ResourceLogsSlice,
 		if accept {
 			continue
 		}
+		// fmt.Println(accept, len(b))
```
Comment (atoulme) [inline exporter/splunkhecexporter/client.go:224]:
```suggestion
```

File: exporter/splunkhecexporter/client.go [hunk 2]
```diff
@@ -241,6 +242,7 @@ func (c *client) pushLogRecords(ctx context.Context, lds plog.ResourceLogsSlice,
 				fmt.Errorf("dropped log event error: event size %d bytes larger than configured max content length %d bytes", len(b), state.bufferMaxLen)))
 			continue
 		}
+		// fmt.Println(accept, len(b), state.rawLength)
```
Comment (atoulme) [inline exporter/splunkhecexporter/client.go:245]:
```suggestion
```

**Review by atoulme — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

File: exporter/splunkhecexporter/buffer.go [hunk 3]
```diff
@@ -71,6 +74,9 @@ func (b *bufferState) Close() error {
 
 // accept returns true if data is accepted by the buffer
 func (b *bufferState) accept(data []byte) (bool, error) {
+	if len(data)+b.rawLength > int(b.maxEventLength) {
```
Comment (atoulme) [inline exporter/splunkhecexporter/buffer.go:77]: this is incorrect. I should have caught this.

Proper would have been:
```
if len(data) > int(b.maxEventLength) {
```

Comment (VihasMakwana) [reply exporter/splunkhecexporter/buffer.go:77]: okay, just to be clear, we're moving from per batch to per event. is that correct?

Comment (atoulme) [reply exporter/splunkhecexporter/buffer.go:77]: correct

**Pull request merged**